### ISSUE TITLE
MiniZinc error: Memory violation detected (segmentation fault).

### ISSUE THREAD

**Issue by SimoPez** [MiniZinc/libminizinc#725]
Hello,
I encountered this error while running a code in the MiniZinc IDE:
`MiniZinc error: Memory violation detected (segmentation fault).
This is a bug. Please file a bug report using the MiniZinc bug tracker.
Process finished with non-zero exit code 6.`

How can I get informations about the reason of the error?

Thank you!

**Comment by Dekker1** [comment]
As the error suggest you've encountered a segmentation fault in the compiler. This is not something that should ever happen, and we would love to receive more information about what exactly caused the error.

When it comes to segmentation faults, we can usually quite easily find it by rerunning the used command (with the accompanying model & data files) in a C++ debugger (`lldb`/`gdb`). Would you be able to provide the command and files? If there is any sensitive information, then feel free to contact me over email instead.

**Comment by SimoPez** [comment]
Thank you very much for your response.

I ran the attached file in the MiniZinc IDE
[hight_p64_k128_o64_r3_Cp_xor_differential_one_solution.txt](https://github.com/MiniZinc/libminizinc/files/12408484/hight_p64_k128_o64_r3_Cp_xor_differential_one_solution.txt)

**Comment by Dekker1** [comment]
This problem was caused by a problem where `SyntaxError` objects would not correctly keep their Location filename alive. The problem in your model has to do with the places where you use a call: `int( ... )`. `int` is not a function MiniZinc. The error that the compiler would have output would have been:
```mzn
Multiple syntax errors:
/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:380.28-30:
constraint w_modadd_0_5[1]=int(100*log2(phi_modadd_0_5[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:382.28-30:
constraint w_modadd_0_5[2]=int(100*log2(phi_modadd_0_5[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:384.28-30:
constraint w_modadd_0_5[3]=int(100*log2(phi_modadd_0_5[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:386.28-30:
constraint w_modadd_0_5[4]=int(100*log2(phi_modadd_0_5[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:388.28-30:
constraint w_modadd_0_5[5]=int(100*log2(phi_modadd_0_5[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:390.28-30:
constraint w_modadd_0_5[6]=int(100*log2(phi_modadd_0_5[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:392.28-30:
constraint w_modadd_0_5[7]=int(100*log2(phi_modadd_0_5[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:414.28-30:
constraint w_modadd_0_7[1]=int(100*log2(phi_modadd_0_7[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:416.28-30:
constraint w_modadd_0_7[2]=int(100*log2(phi_modadd_0_7[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:418.28-30:
constraint w_modadd_0_7[3]=int(100*log2(phi_modadd_0_7[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:420.28-30:
constraint w_modadd_0_7[4]=int(100*log2(phi_modadd_0_7[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:422.28-30:
constraint w_modadd_0_7[5]=int(100*log2(phi_modadd_0_7[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:424.28-30:
constraint w_modadd_0_7[6]=int(100*log2(phi_modadd_0_7[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:426.28-30:
constraint w_modadd_0_7[7]=int(100*log2(phi_modadd_0_7[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:448.28-30:
constraint w_modadd_0_9[1]=int(100*log2(phi_modadd_0_9[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:450.28-30:
constraint w_modadd_0_9[2]=int(100*log2(phi_modadd_0_9[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:452.28-30:
constraint w_modadd_0_9[3]=int(100*log2(phi_modadd_0_9[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:454.28-30:
constraint w_modadd_0_9[4]=int(100*log2(phi_modadd_0_9[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:456.28-30:
constraint w_modadd_0_9[5]=int(100*log2(phi_modadd_0_9[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:458.28-30:
constraint w_modadd_0_9[6]=int(100*log2(phi_modadd_0_9[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:460.28-30:
constraint w_modadd_0_9[7]=int(100*log2(phi_modadd_0_9[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:482.29-31:
constraint w_modadd_0_11[1]=int(100*log2(phi_modadd_0_11[1]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:484.29-31:
constraint w_modadd_0_11[2]=int(100*log2(phi_modadd_0_11[2]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:486.29-31:
constraint w_modadd_0_11[3]=int(100*log2(phi_modadd_0_11[3]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:488.29-31:
constraint w_modadd_0_11[4]=int(100*log2(phi_modadd_0_11[4]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:490.29-31:
constraint w_modadd_0_11[5]=int(100*log2(phi_modadd_0_11[5]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:492.29-31:
constraint w_modadd_0_11[6]=int(100*log2(phi_modadd_0_11[6]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:494.29-31:
constraint w_modadd_0_11[7]=int(100*log2(phi_modadd_0_11[7]));
                            ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:872.28-30:
constraint w_modadd_1_1[1]=int(100*log2(phi_modadd_1_1[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:874.28-30:
constraint w_modadd_1_1[2]=int(100*log2(phi_modadd_1_1[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:876.28-30:
constraint w_modadd_1_1[3]=int(100*log2(phi_modadd_1_1[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:878.28-30:
constraint w_modadd_1_1[4]=int(100*log2(phi_modadd_1_1[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:880.28-30:
constraint w_modadd_1_1[5]=int(100*log2(phi_modadd_1_1[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:882.28-30:
constraint w_modadd_1_1[6]=int(100*log2(phi_modadd_1_1[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:884.28-30:
constraint w_modadd_1_1[7]=int(100*log2(phi_modadd_1_1[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:906.28-30:
constraint w_modadd_1_3[1]=int(100*log2(phi_modadd_1_3[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:908.28-30:
constraint w_modadd_1_3[2]=int(100*log2(phi_modadd_1_3[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:910.28-30:
constraint w_modadd_1_3[3]=int(100*log2(phi_modadd_1_3[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:912.28-30:
constraint w_modadd_1_3[4]=int(100*log2(phi_modadd_1_3[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:914.28-30:
constraint w_modadd_1_3[5]=int(100*log2(phi_modadd_1_3[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:916.28-30:
constraint w_modadd_1_3[6]=int(100*log2(phi_modadd_1_3[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:918.28-30:
constraint w_modadd_1_3[7]=int(100*log2(phi_modadd_1_3[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:940.28-30:
constraint w_modadd_1_5[1]=int(100*log2(phi_modadd_1_5[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:942.28-30:
constraint w_modadd_1_5[2]=int(100*log2(phi_modadd_1_5[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:944.28-30:
constraint w_modadd_1_5[3]=int(100*log2(phi_modadd_1_5[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:946.28-30:
constraint w_modadd_1_5[4]=int(100*log2(phi_modadd_1_5[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:948.28-30:
constraint w_modadd_1_5[5]=int(100*log2(phi_modadd_1_5[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:950.28-30:
constraint w_modadd_1_5[6]=int(100*log2(phi_modadd_1_5[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:952.28-30:
constraint w_modadd_1_5[7]=int(100*log2(phi_modadd_1_5[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:974.28-30:
constraint w_modadd_1_7[1]=int(100*log2(phi_modadd_1_7[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:976.28-30:
constraint w_modadd_1_7[2]=int(100*log2(phi_modadd_1_7[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:978.28-30:
constraint w_modadd_1_7[3]=int(100*log2(phi_modadd_1_7[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:980.28-30:
constraint w_modadd_1_7[4]=int(100*log2(phi_modadd_1_7[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:982.28-30:
constraint w_modadd_1_7[5]=int(100*log2(phi_modadd_1_7[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:984.28-30:
constraint w_modadd_1_7[6]=int(100*log2(phi_modadd_1_7[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:986.28-30:
constraint w_modadd_1_7[7]=int(100*log2(phi_modadd_1_7[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1364.28-30:
constraint w_modadd_2_1[1]=int(100*log2(phi_modadd_2_1[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1366.28-30:
constraint w_modadd_2_1[2]=int(100*log2(phi_modadd_2_1[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1368.28-30:
constraint w_modadd_2_1[3]=int(100*log2(phi_modadd_2_1[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1370.28-30:
constraint w_modadd_2_1[4]=int(100*log2(phi_modadd_2_1[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1372.28-30:
constraint w_modadd_2_1[5]=int(100*log2(phi_modadd_2_1[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1374.28-30:
constraint w_modadd_2_1[6]=int(100*log2(phi_modadd_2_1[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1376.28-30:
constraint w_modadd_2_1[7]=int(100*log2(phi_modadd_2_1[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1398.28-30:
constraint w_modadd_2_3[1]=int(100*log2(phi_modadd_2_3[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1400.28-30:
constraint w_modadd_2_3[2]=int(100*log2(phi_modadd_2_3[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1402.28-30:
constraint w_modadd_2_3[3]=int(100*log2(phi_modadd_2_3[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1404.28-30:
constraint w_modadd_2_3[4]=int(100*log2(phi_modadd_2_3[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1406.28-30:
constraint w_modadd_2_3[5]=int(100*log2(phi_modadd_2_3[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1408.28-30:
constraint w_modadd_2_3[6]=int(100*log2(phi_modadd_2_3[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1410.28-30:
constraint w_modadd_2_3[7]=int(100*log2(phi_modadd_2_3[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1432.28-30:
constraint w_modadd_2_5[1]=int(100*log2(phi_modadd_2_5[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1434.28-30:
constraint w_modadd_2_5[2]=int(100*log2(phi_modadd_2_5[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1436.28-30:
constraint w_modadd_2_5[3]=int(100*log2(phi_modadd_2_5[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1438.28-30:
constraint w_modadd_2_5[4]=int(100*log2(phi_modadd_2_5[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1440.28-30:
constraint w_modadd_2_5[5]=int(100*log2(phi_modadd_2_5[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1442.28-30:
constraint w_modadd_2_5[6]=int(100*log2(phi_modadd_2_5[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1444.28-30:
constraint w_modadd_2_5[7]=int(100*log2(phi_modadd_2_5[7]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1466.28-30:
constraint w_modadd_2_7[1]=int(100*log2(phi_modadd_2_7[1]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1468.28-30:
constraint w_modadd_2_7[2]=int(100*log2(phi_modadd_2_7[2]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1470.28-30:
constraint w_modadd_2_7[3]=int(100*log2(phi_modadd_2_7[3]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1472.28-30:
constraint w_modadd_2_7[4]=int(100*log2(phi_modadd_2_7[4]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1474.28-30:
constraint w_modadd_2_7[5]=int(100*log2(phi_modadd_2_7[5]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1476.28-30:
constraint w_modadd_2_7[6]=int(100*log2(phi_modadd_2_7[6]));
                           ^^^
Error: syntax error, unexpected int

/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/github_725.mzn:1478.28-30:
constraint w_modadd_2_7[7]=int(100*log2(phi_modadd_2_7[7]));
                           ^^^
Error: syntax error, unexpected int
```

The fix for this problem will be merged into `develop` shortly

**Comment by Dekker1** [comment]
Although I must have messed up the message during an amend in the development process, 4bfad6399dab00e4260fde75e272d82b02b33525 will resolve the issue.

**Comment by SimoPez** [comment]
Thank you very much! It worked

**Comment by SimoPez** [comment]
Hello!
I have fixed some minor bugs in my model after your resolution of the first problem and now I encountered the same error again.
I was wandering if you could help me again with this.
[hight_p64_k128_o64_r3_Cp_xor_differential_one_solution.txt](https://github.com/MiniZinc/libminizinc/files/12505949/hight_p64_k128_o64_r3_Cp_xor_differential_one_solution.txt)
Thank you very much and sorry for bothering you

**Comment by Dekker1** [comment]
Although I'm happy to run it through the `develop` branch, it seems that the error might now be hidden by the missing include that you haven't shared. Currently the compiler outputs:
```mzn
Warning: variable `k` shadows variable with the same name in line 4.36
/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/share/minizinc/std/fzn_geost.mzn:12.45-48

Error: type error: no function or predicate with name `LShift' found
/Users/dekker1/Library/Mobile Documents/com~apple~CloudDocs/Code/github.com/MiniZinc/libminizinc/test.mzn:119.49-74
```

**Comment by cyderize** [comment]
This should now be fixed in 2.8.0
